src/python/error.rs

```diff
@@ -1,9 +1,13 @@
-use pyo3::{exceptions::PyValueError, PyErr};
+use pyo3::{exceptions::PyTypeError, exceptions::PyValueError, PyErr};
 
 use crate::error::DaftError;
 
 impl std::convert::From<DaftError> for PyErr {
     fn from(err: DaftError) -> PyErr {
-        PyValueError::new_err(err.to_string())
+        match err {
+            DaftError::ExprResolveTypeError { .. } => PyTypeError::new_err(err.to_string()),
```

Comment:
Why do we have this match block? All the branches seem to do the same thing

Reply:
These propagate a TypeError instead of a ValueError, though I should probably combine it with the block below.

Reply:
Combined as a single match block